eng/scripts/Language-Settings.ps1

```diff
@@ -0,0 +1,92 @@
+$Language = "rust"
+$LanguageDisplayName = "Rust"
+$PackageRepository = "crates.io"
+$packagePattern = "Cargo.toml"
+$MetadataUri = "https://raw.githubusercontent.com/Azure/azure-sdk/main/_data/releases/latest/rust-packages.csv"
+$GithubUri = "https://github.com/Azure/azure-sdk-for-rust"
+$PackageRepositoryUri = "https://crates.io/crates"
+
+function Get-AllPackageInfoFromRepo ([string] $ServiceDirectory) {
+    $allPackageProps = @()
+    Push-Location $RepoRoot
+    try {
+        $searchPath = 'sdk'
+
+        if ($ServiceDirectory -and $ServiceDirectory -ne 'auto') {
+            $searchPath = Join-Path 'sdk' $ServiceDirectory
+        }
+
+        $cargoTomlFiles = Get-ChildItem -Path $searchPath -Recurse -Filter 'Cargo.toml' -ErrorAction SilentlyContinue
```

Comment:
I think this is too much of an assumption. Cargo doesn't yet support nested workspaces, but it's being discussed and would solve a number of problems, so when it is supported I'd love to adopt it then this would fail. Instead, if you want to get all packages, perhaps `cargo metadata` would be better.

```pwsh
$packages = cargo metadata --format-version 1 | convertfrom-json | select -expand packages | where { $_.manifest_path.StartsWith((join-path $PWD 'sdk')) }
foreach ($package in $packages) {
    $manifest = cargo read-manifest --manifest-path $package.manifest_path
    # ...
}
```

Reply:
I updated to use `cargo metadata`